index.js

```diff
@@ -6,13 +6,19 @@ var through = require("through2");
 // # Filter
 // Create a through stream that only passes data that passes the given test
 // function
-module.exports = function(test, opt) {
+var filter = module.exports = function(test, opt) {
 	return through(opt || {}, function(data, _, cb) {
 		if (test(data)) cb(null, data);
 		else cb(null);
 	});
 };
 
+module.exports.obj = function(test, opt) {
+	opt = opt || {}
+	opt.objectMode = true
```

Comment:
By setting `opt.objectMode = true` it will change the object passed in. Honestly not sure about the `highWaterMark` -- probably because the default is 16k (bytes in a non-object stream) and that doesn't make sense with objects -- was just trying to stick to `through2`'s semantics.